skcosmo/preprocessing/FlexibleScaler.py

```diff
@@ -0,0 +1,242 @@
+    :type ref_mean: array
+    """
+
+    def __init__(self, *, ref_cmean=None, ref_rmean=None, ref_mean=None):
+        """Initialize KernelFlexibleCenterer."""
+        pass
+
+    def fit(self, K=None, y=None, ref_cmean=None, ref_rmean=None, ref_mean=None):
+        """Fit KernelFlexibleCenterer
+
+        :param K: Kernel matrix
+        :type K: ndarray of shape (n_samples, n_samples)
+        :param y: ignored
+        :param ref_cmean: an array with means for each column.
+        :type ref_cmean: array of shape (insert shape here)
+        :param ref_rmean: an array with means for each row.
+        :type ref_rmean: array of shape (insert shape here)
+        :param ref_mean: an average for the whole kernel matrix
+        :type ref_mean: array
+        :return: itself
+        """
+        if K is not None:
+            if K.shape[0] != K.shape[1]:
+                raise ValueError(
+                    "The reference kernel is not square, and does not define a RKHS"
+                )
+        else:
+            assert (
+                ref_cmean is not None and ref_rmean is not None and ref_mean is not None
+            )
+
+            if len(ref_rmean) != len(ref_cmean):
+                raise ValueError(
+                    "The reference kernel is not square, and does not define a RKHS"
+                )
+        self.ref_cmean = ref_cmean
+        self.ref_rmean = ref_rmean
+        self.ref_mean = ref_mean
+        self.reference_shape = []
+        if ref_rmean is None:
+            self.reference_shape.append(K.shape[0])
+        else:
+            self.reference_shape.append(len(ref_rmean))
+        if ref_cmean is None:
+            self.reference_shape.append(K.shape[1])
+        else:
+            self.reference_shape.append(len(ref_cmean))
+        self.reference_shape = tuple(self.reference_shape)
+        if self.ref_cmean is None:
+            self.ref_cmean = K.mean(axis=0)
+        if self.ref_rmean is None:
+            self.ref_rmean = K.mean(axis=1)
+        if self.ref_mean is None:
+            self.ref_mean = K.mean()
+
+        return self
+
+    def transform(self, K, y=None):
+        """Centering our Kernel. Previously you should fit data.
+
+        :param K: Kernel matrix
+        :type K: ndarray of shape (n_samples, n_samples)
+        :param y: ignored
+        :return: tranformed matrix Kc
+        """
+        parameters = self.get_params()
+        """check each of the parameters self.reference, self.ref_mean, self.ref_cmean, 
+        and self.ref_rmean.They must all be non - None
+        """
+        for key in parameters.keys():
+            if parameters[key] is None:
+                raise sk.exceptions.NotFittedError(
+                    "This "
+                    + type(self).__name__
+                    + " instance is not fitted yet. Call 'fit' with appropriate arguments before using this estimator."
+                )
+        if K.shape != self.reference_shape:
```

Comment:
it's only necessary that their latter dimension be the same, so:
```suggestion
        if K.shape[1] != self.reference_shape[1]:
```